src/services/base-filters-parser.js

```diff
@@ -0,0 +1,74 @@
+import _ from 'lodash';
+import { InvalidFiltersFormatError } from './error';
+
+// NOTICE: Parse the given filters into a valid JSON.
+const parseFiltersString = (filtersString) => {
+  try {
+    return filtersString ? JSON.parse(filtersString) : null;
+  } catch (error) {
+    throw new InvalidFiltersFormatError('Invalid filters JSON format');
+  }
+};
+
+// NOTICE: Apply the formatCondition function to a condition (leaf).
+const parseCondition = (condition, formatCondition) => {
+  if (_.isEmpty(condition)) { throw new InvalidFiltersFormatError('Empty condition in filter'); }
+
```

Comment:
You should check for the format here: It should be an object with the 3 keys `field`, `operator` and `value`